pkg/machinery/config/types/v1alpha1/v1alpha1_stability_test.go

```diff
@@ -0,0 +1,121 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package v1alpha1_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/siderolabs/talos/pkg/machinery/config"
+	"github.com/siderolabs/talos/pkg/machinery/config/configpatcher"
+	"github.com/siderolabs/talos/pkg/machinery/config/encoder"
+	"github.com/siderolabs/talos/pkg/machinery/config/generate"
+	"github.com/siderolabs/talos/pkg/machinery/config/generate/secrets"
+	"github.com/siderolabs/talos/pkg/machinery/config/machine"
+	"github.com/siderolabs/talos/pkg/machinery/config/types/v1alpha1"
+)
+
+// TestConfigEncodingStability ensures that the encoding of a configuration is stable as we moved forward with the config format.
+func TestConfigEncodingStability(t *testing.T) {
+	t.Parallel()
+
+	// flip this to generate missing configs
+	const generateMode = false
+
+	secretsBundle, err := secrets.LoadBundle("testdata/stability/secrets.yaml")
+	require.NoError(t, err)
+
+	for _, versionContract := range []*config.VersionContract{
+		config.TalosVersion1_3,
+		config.TalosVersion1_4,
+		config.TalosVersion1_5,
+		config.TalosVersion1_6,
```

Comment:
should we make this fail explicitly if there's a newer version untested? so we don't forget it in the future?

Reply:
I can do that, good point

Reply:
added a check